src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -1575,7 +1548,7 @@ void CodeGen::codeGenInstruction(ByteCodeOp operation)
 
             if (descriptor.returnType != FieldType(BaseType::Void))
             {
-                operandStack.push_back(call);
+                operandStack.push_back(extendToStackType(builder, descriptor.returnType, call));
```

Comment:
We should definitely add regression tests for these (doesn't have to be this PR)

Reply:
We do have test for this case (e.g. `assert.java`)

Reply:
Probably more accidental than anything, but good to know